### ISSUE TITLE
UKUU-GTK crashes at start-up

### ISSUE THREAD

**Issue by jdferron** [teejee2008/ukuu-plus#34]
$ ukuu-gtk
[13:00:39.362] ukuu-gtk v19.07.5
[13:00:39.395] [Warning] Deleted invalid lock
[13:00:39.487] Temp: /tmp/ukuu-gtk_cXBIxUN3
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.956: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.956: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.956: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.956: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.956: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.956: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.956: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.956: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.956: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.956: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.957: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.957: json_object_get_boolean_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.957: json_object_get_boolean_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.957: json_object_get_boolean_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.957: json_object_get_boolean_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.957: json_object_get_string_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.957: json_object_get_array_member: assertion 'node != NULL' failed
(ukuu-gtk:13963): Json-CRITICAL **: 13:00:41.957: json_array_get_elements: assertion 'array != NULL' failed
(ukuu-gtk:13963): GLib-CRITICAL **: 13:00:41.987: g_strsplit: assertion 'string != NULL' failed
Segmentation fault (core dumped)
$

**Comment by teejee2008** [comment]
Update to v19.10.1

**Comment by abylikhsanov** [comment]
@teejee2008 I have a similar problem (crashesh at the startup) but I have a message:
```
[07:31:04.942] ukuu-gtk v19.10.1
[07:31:05.218] Temp: /tmp/ukuu-gtk_GWDlKLPJ
[07:31:07.304] E: Kernel list is empty!!
```
I am on kernel 4.15
Note that on a free version it was able to find kernel 5.3x for the update

**Comment by teejee2008** [comment]
It is not able to connect to internet. Are you behind a VPN?

**Comment by Random-Chain** [comment]
I'm getting the same "list is empty" error.
ukuu-gtk v19.10.1, on Pop!_OS 19.10, kernel 5.3.0-7625-generic, if it's relevant.
Not behind VPN, tried starting it on several different networks.
It worked before on this machine, but now stopped for some reason.

Is there any log or debug I can provide?

**Comment by timrichardson** [comment]
I have newly installed the paid version. A blank windows appears for a few seconds, and then it closes. KDE Neon. There are  no connectivity issues and there is no VPN.

Run from the command line:

```
tim@moncrief:~$ ukuu-gtk
[08:07:32.836] ukuu-gtk v19.10.1
[08:07:32.943] Temp: /tmp/ukuu-gtk_nFNikTwx
[08:07:34.611] E: Kernel list is empty!!
tim@moncrief:~$ 

```
```
tim@moncrief:~$ ukuu --list

(process:30461): GLib-CRITICAL **: 08:03:55.720: g_str_has_prefix: assertion 'prefix != NULL' failed

(process:30461): GLib-CRITICAL **: 08:03:55.720: g_str_has_prefix: assertion 'prefix != NULL' failed

(process:30461): GLib-CRITICAL **: 08:03:55.720: g_str_has_prefix: assertion 'prefix != NULL' failed
tim@moncrief:~$ 
```

**Comment by teejee2008** [comment]
Will investigate on PopOS/KDE Neon and let you know

**Comment by RobbinBaauw** [comment]
I have the same problem, also using PopOS 19.10 but with Gnome.

`ukuu --list` doesn't throw any errors for me, it just returns nothing.

**Comment by teejee2008** [comment]
I just tested on a fresh install of PopOS 19.10 and KDE Neon and it's working as expected.

It seems that the paid version is either not installed properly or the free version is also installed. Please note that installing both the free and paid versions will cause problems.

Try the following:

Remove all installed versions of Ukuu
`sudo apt purge ukuu*`

Remove the cache and configs
`sudo rm -rf  ~/.cache/ukuu ~/.config/ukuu`

Reinstall the paid version by running the install script.

If you still get errors then run the following and send me the complete output by email.
`sudo aptik --list --debug`

**Comment by teejee2008** [comment]
Issue with PopOS 19.10 is fixed in v19.12. Please update.
```
sudo apt update
sudo apt install ukuu*
```
Issue was related to the latest PopOS kernel (5.3.0.27) which has package version 5.3.0-7625 instead of 5.3.0-27. If you were running this kernel then Ukuu would fail to start as it didn't know which installed packages belong to the running kernel.

**Comment by RobbinBaauw** [comment]
Thanks! It works now!

**Comment by timrichardson** [comment]
I currently have a custom kernel to work around a bad backport of a security bug which has hurt power saving. ukuu doesn't start, as you say. Can you change the behaviour so it can start? Because it's a bit limiting if we can't have custom kernel builds.

**Comment by teejee2008** [comment]
If it's a custom kernel it won't be able to determine the version. It needs to know the version so that it can display the running kernel in the list, notify/install updates, and prevent the user from uninstalling the running kernel.

For now, you can boot into an official or mainline kernel and then use Ukuu. I'll see if I can add a workaround for this situation in a future update.

**Comment by teejee2008** [comment]
Fixed in v20.05
